In [1]:
# Hypothesis Testing

## Testing Whether Weekly Operating Costs Are Higher Than the Theoretical Model

In [2]:
# Given information

sample_mean = 3050
sample_size = 25

# Number of units produced per week
mu_units = 600
sigma_units = 25

# Cost model: W = 1000 + 5X
theoretical_mean = 1000 + (5 * mu_units)

# Standard deviation of weekly operating cost
sigma_cost = 5 * sigma_units

print("Sample Mean Weekly Cost:", sample_mean)
print("Sample Size:", sample_size)
print("Theoretical Mean Weekly Cost:", theoretical_mean)
print("Standard Deviation of Weekly Cost:", sigma_cost)

Sample Mean Weekly Cost: 3050
Sample Size: 25
Theoretical Mean Weekly Cost: 4000
Standard Deviation of Weekly Cost: 125


## Step 1: Hypothesis Statement

The restaurant owners claim that the actual mean weekly operating cost is higher than the theoretical mean weekly operating cost.

**Null Hypothesis (H₀):** μ = Rs. 4,000

**Alternative Hypothesis (H₁):** μ > Rs. 4,000

This is a right-tailed hypothesis test because the claim is that the weekly operating cost is higher than the theoretical value.

In [5]:
# Step 2: Calculate the z test statistic

import numpy as np

z_statistic = (sample_mean - theoretical_mean) / (sigma_cost / np.sqrt(sample_size))

print("Z Test Statistic:", z_statistic)

Z Test Statistic: -38.0


In [7]:
pip install scipy

  Using cached scipy-1.18.1-cp313-cp313-macosx_14_0_arm64.whl.metadata (62 kB)
Using cached scipy-1.18.1-cp313-cp313-macosx_14_0_arm64.whl (20.5 MB)

[notice] A new release of pip is available: 26.0.1 -> 26.2.1
[notice] To update, run: pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.


In [8]:
# Step 3: Calculate the p-value and critical value

from scipy import stats

alpha = 0.05

# Right-tailed p-value
p_value = 1 - stats.norm.cdf(z_statistic)

# Critical z-value for a right-tailed test
critical_value = stats.norm.ppf(1 - alpha)

print("Significance Level (alpha):", alpha)
print("P-value:", p_value)
print("Critical Value:", critical_value)

Significance Level (alpha): 0.05
P-value: 1.0
Critical Value: 1.644853626951472


In [9]:
# Step 4: Make the decision

if z_statistic > critical_value:
    decision = "Reject the null hypothesis (H0)"
else:
    decision = "Fail to reject the null hypothesis (H0)"

print("Decision:", decision)

Decision: Fail to reject the null hypothesis (H0)


## Step 5: Conclusion

At the 5% significance level, the p-value is 1.0, which is greater than the significance level of 0.05. Also, the calculated z test statistic of -38.0 is less than the critical value of 1.645.

Therefore, we fail to reject the null hypothesis.

There is not enough statistical evidence to support the restaurant owners' claim that the mean weekly operating cost is higher than the theoretical cost suggested by the model.S